astropy/io/ascii/tests/test_read.py

```diff
@@ -1034,24 +1034,40 @@ def test_almost_but_not_quite_daophot():
     assert len(dat) == 3
 
 
-@pytest.mark.parametrize('fast', [True, False])
+@pytest.mark.parametrize('fast', [True, False, 'force'])
 def test_commented_header_comments(fast):
     """
     Test that comments in commented_header are as expected and that the
     table round-trips.
     """
+    comments = ['comment 1', 'comment 2', 'comment 3']
     lines = ['# a b',
              '# comment 1',
              '# comment 2',
+             '# comment 3',
              '1 2',
              '3 4']
     dat = ascii.read(lines, format='commented_header', fast_reader=fast)
-    assert dat.meta['comments'] == ['comment 1', 'comment 2']
+    assert dat.meta['comments'] == comments
 
     out = StringIO()
     ascii.write(dat, out, format='commented_header', fast_writer=fast)
     assert out.getvalue().splitlines() == lines
 
+    lines.insert(1, lines.pop(0))
+    dat = ascii.read(lines, format='commented_header', header_start=1, fast_reader=fast)
+
+    lines.insert(2, lines.pop(1))
+    dat = ascii.read(lines, format='commented_header', header_start=2, fast_reader=fast)
+    assert dat.meta['comments'] == comments
+    dat = ascii.read(lines, format='commented_header', header_start=-2, fast_reader=fast)
+    assert dat.meta['comments'] == comments
+
+    lines.insert(3, lines.pop(2))
+    dat = ascii.read(lines, format='commented_header', header_start=-1, fast_reader=fast)
+    assert dat.meta['comments'] == comments
```

Comment:
These two tests look good, so they should both be done after all four of the test reads.

Reply:
You mean check `colnames` after each read? This already seemed extra-cautious, but in theory of course the reader might end up creating columns like `comment`, `1` **and** stripping the wrong comment line at the same time...

Reply:
Oh, I see one
`assert dat.meta['comments'] == comments`
got lost when adding the extra tests for `header_start` = ±2.